packages/freezed/test/json_test.dart

```diff
@@ -28,10 +29,38 @@ Future<void> main() async {
       DefaultValue(42),
     );
   });
-  test('@Default does not imply a @JsonKey if one is already specified', () {
+  test('@required implies a @JsonKey', () {
+    final value = RequiredValue(value: 42);
+    expect(
+      value.toJson(),
+      {'value': 42},
+    );
+
+    expect(
+      () => RequiredValue.fromJson(<String, dynamic>{}),
+      throwsA(isA<CheckedFromJsonException>()),
+    );
+
+    expect(
+      () => RequiredValue.fromJson(<String, dynamic>{'value': null}),
+      throwsA(isA<CheckedFromJsonException>()),
+    );
+  });
+  test('@nullable implies a @JsonKey', () {
```

Comment:
I don't understand what this test does, since this is already the current behavior

Reply:
Fair, maybe this is not the way to test this. I just wanted to make sure that if we had the `@nullable` decorator that we would add the corresponding decorator: `@JsonKey(required: false, disallowNullValue: false)`

Do you have any suggestions on how I can test this?

Reply:
For added context, the reason why I think this is important is because someone could write:
```dart
factory User({@nullable @JsonKey(disallowNullValue: true) int age})
```
Resulting again in an indeterministic behavior

Reply:
The behavior is fixed: `@JsonKey` inside the constructor explicitly takes over every other configuration.

Reply:
But don't you think that this is problematic when working in a big team, and looking at someone else's code that you're not familiar with?

Although the behavior might be fixed based on what happens at run time, when reading this code it is not immediately clear which one of those decorators should superseed the other. I think it might be nice to prevent users from defining decorators that could in theory contradict each other, which means that one of them would actually end up being ignored.

Reply:
To me, it doesn't matter, as people shouldn't need to specific `@JsonKey` to begin with.

The support for `@JsonKey` in parameters is here only for people want to override the default behavior. So in that situation, it is clear that it takes over everything else.

What you want is not to check for inconsistencies. You want to change the default behavior.

Reply:
But there's also the case where you might want to specify a `@JsonKey` just so that you get a `CheckedFromJsonException` instead of a runtime assertion error. (Although if we modify our constructors the way you suggested above then this wouldn't matter anymore)

I do think it might still be nice to throw a `InvalidGenerationSourceError` during codegen when you have defined decorators that contradict each other as a sanity check? What do you think of that?

Reply:
> I do think it might still be nice to throw a InvalidGenerationSourceError during codegen when you have defined decorators that contradict each other as a sanity check? What do you think of that?

I would agree if people were expected to write `@JsonKey(something: true)` themselves. But they are not
As I mentioned, `@JsonKey` on parameters is only here for edge-cases to let people have fine grained control over the annotation.

Reply:
> As I mentioned, @JsonKey on parameters is only here for edge-cases to let people have fine grained control over the annotation.

Of course, yeah. I just meant to prevent errors from slipping through when dealing with these edge-cases.